hypersync-format/src/types/quantity.rs

```diff
@@ -146,7 +146,7 @@ pub fn decode_hex(value: &str) -> Result<Vec<u8>> {
         .strip_prefix("0x")
         .ok_or_else(|| Error::InvalidHexPrefix(value.to_owned()))?;
 
-    if val.is_empty() || val.starts_with('0') {
+    if val.is_empty() {
```

Comment:
No additional overhead on the standard hex repr since this check is removed and simply done on  the bytes after decode